qiskit_experiments/data_processing/nodes.py

```diff
@@ -0,0 +1,244 @@
+            new_mem = []
+            for shot in data["memory"]:
+                new_mem.append([self.scale * _[0] for _ in shot])
+
+            if self.average:
+                new_mem = list(np.mean(np.array(new_mem), axis=0))
+
+        # Averaged data
+        else:
+            new_mem = [self.scale * _[0] for _ in data["memory"]]
+
+        data[self.node_output] = new_mem
+
+
+@iq_data
+@prev_node(Kernel)
+class ToImag(DataAction):
+    """IQ data post-processing. This returns imaginary part of IQ data."""
+
+    def __init__(self, scale: Optional[float] = 1.0, average: bool = False):
+        """
+        Args:
+            scale: scale by which to multiply the imaginary part of the data.
+        """
+        self.scale = scale
+        self.average = average
+        super().__init__()
+
+    @property
+    def node_output(self) -> str:
+        """Key under which ToImag stores the data."""
+        return "memory_imag"
+
+    @property
+    def node_inputs(self) -> List[str]:
+        """Returns a list of input data that the node can process."""
+        return ["memory"]
+
+    def process(self, data: Dict[str, Any]):
+        """
+        Scales the imaginary part of IQ data.
+
+        Args:
+            data: The data dict. IQ data is stored under memory.
+
+        Raises:
+            DataProcessorError: if the data does not contain memory.
+        """
+
+        # Single shot data
+        if isinstance(data["memory"][0][0], list):
+            new_mem = []
+            for shot in data["memory"]:
+                new_mem.append([self.scale * _[1] for _ in shot])
+
+            if self.average:
+                new_mem = list(np.mean(np.array(new_mem), axis=0))
+
+        # Averaged data
+        else:
+            new_mem = [self.scale * _[0] for _ in data["memory"]]
+
+        data[self.node_output] = new_mem
+
+
+@population
+@prev_node(Discriminator)
+class Population(DataAction):
+    """Count data post processing. This returns population."""
+
+    @property
+    def node_output(self) -> str:
+        """Key under which Population stores the data."""
+        return "populations"
+
+    @property
+    def node_inputs(self) -> List[str]:
+        """Returns a list of input data that the node can process."""
+        return ["counts"]
+
+    def process(self, data: Dict[str, Any]):
+        """
+        Args:
+            data: The data dictionary. This will modify the dict in place,
+                taking the data under counts and adding the corresponding
+                populations.
+
+        Raises:
+            DataProcessorError: if counts are not in the given data.
+        """
+
+        counts = data.get("counts")
+
+        populations = np.zeros(len(list(counts.keys())[0]))
+
+        shots = 0
+        for bit_str, count in counts.items():
+            shots += 1
+            for ind, bit in enumerate(bit_str):
+                if bit == "1":
+                    populations[ind] += count
+
```

Comment:
This function will replace `level2_probability` in #19, and should be more similar to `level2_probability`:
1. Parameter `outcome` - not necessarily bit 1 is the one that increases the population.
2. Only one population, that can be of multiple qubits: Consider for example a randomized benchmarking experiment that's done in parallel for qubit 1 and qubits 2 and 3 (qubits 2 and 3 are grouped together, and are not independent, they are independent from qubit 1). We program this scenario with a parallel experiment, which consists of two RB experiments, one for qubit 1 and the other one for qubits 2 and 3. The current code of `process` does not support the grouping of qubits 2 and 3 (it will create separate populations for each qubit). Moreover, these separate populations are in fact not needed, because independent qubits are handled by parallel experiments.